examples/docker/gpt_neo_docker.py

```diff
@@ -0,0 +1,76 @@
+"""
+Pipline has support for creating FastAPI endpoints for your Pipelines out of the box.
+This file demonstrates this by creating a basic GPT-Neo Pipeline with a custom
+environment and deploying the docker image.
+
+Run this code in this directory:
+
+python gpt_neo_docker.py
+
+and then start the docker containers created:
+
+sudo docker compose up -d
+
+you can test the running images:
+
+
+curl --request POST \
+  --url http://localhost:5010/v2/run \
+  --header 'Content-Type: application/json' \
+  --data '{"pipeline_id":"gptneo", "data":"Hey there my name is"}'
+
+"""
+
+
+from pipeline import Pipeline, Variable, docker, pipeline_function, pipeline_model
+from pipeline.objects.environment import Environment
+
+
+@pipeline_model
+class PipelineGPTNeo:
+    def __init__(self):
+        self.model = None
+        self.tokenizer = None
+
+    @pipeline_function
+    def predict(self, input_data: str) -> str:
+        if self.model is None:
+            from transformers import GPT2Tokenizer, GPTNeoForCausalLM
+
+            self.model = GPTNeoForCausalLM.from_pretrained("EleutherAI/gpt-neo-125M")
+            self.tokenizer = GPT2Tokenizer.from_pretrained("EleutherAI/gpt-neo-125M")
+
+        input_ids = self.tokenizer(input_data, return_tensors="pt").input_ids
+
+        gen_tokens = self.model.generate(
+            input_ids,
+            do_sample=True,
+            temperature=0.9,
+            max_length=100,
+        )
+        gen_text = self.tokenizer.batch_decode(gen_tokens)[0]
+        return gen_text
+
+
+with Pipeline("GPTNeo") as builder:
+    in_1 = Variable(str, is_input=True)
```

Comment:
Just an aside, but seeing as this is a PR about cleanup: is `is_input` necessary here? Would a user ever create a output variable explicitly, e.g. I guess via `Variable(int, is_input=False)`? Could consider making `is_input=True` the default.

Reply:
They're mutually exclusive as the `is_output` field can be true even if the `is_input` field is true. 
These field are going to be removed in a future PR altogether and the context manager will set them instead of the user, it's so ugly at the moment. I'm happy with this example as is for the moment